modules/promotion/src/PromotionStorage.php

```diff
@@ -107,6 +107,46 @@ public function loadByCoupon(OrderTypeInterface $order_type, StoreInterface $sto
   }
 
   /**
+   * Builds a query that will load all promotions that are no longer valid
+   * determined by the base field limiting values.
+   *
+   * @param bool $only_enabled
+   *   Only include currently enabled promotions that have expired.
+   * @return array|bool|\Drupal\Core\Entity\EntityInterface[]
+   *   The expired promotion entities. Returns FALSE if none found.
+   */
+  public function loadExpired($only_enabled = TRUE) {
+    // Subquery to determine the amount of times a coupon has been used.
+    $usage_query = $this->database->select('commerce_promotion_usage', 'cpu');
+    $usage_query->addExpression('COUNT(cpu.promotion_id)', 'count');
+    $usage_query->where('cpu.promotion_id = cpd.promotion_id');
+    $usage_query->groupBy('cpu.promotion_id');
```

Comment:
We already have this query in the usage service `:: getUsageMultiple`. It feels like we should rework this method a bit so it can be used.

https://github.com/drupalcommerce/commerce/blob/8.x-2.x/modules/promotion/src/PromotionUsage.php#L65